src/+replab/+fp/CosetTable.m

```diff
@@ -544,10 +544,14 @@ function scanAndFill(self, alpha, w)
                 end
                 if j < i
                     self.coincidence(f, b);
-                    f = alpha;
-                    i = 1;
-                    b = alpha;
-                    j = r;
+                    if self.p(alpha) == alpha
```

Comment:
I've looked at the appropriate course of action there, I believe we should just return. There is probably a typo in the book, which is not acknowledged in their erratum. Let's discuss.